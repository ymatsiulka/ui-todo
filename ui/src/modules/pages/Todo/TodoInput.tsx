import React from 'react';
import { Input } from 'modules/shared';
import { input } from 'appConstants';
// import styles from './TodoInput.m.scss';

interface TodoInputProps {
  value: string;
  onChangeHandler: React.ChangeEventHandler<HTMLInputElement>;
  onKeyUpHandler: React.KeyboardEventHandler<HTMLInputElement>;
}

const TodoInput: React.FC<TodoInputProps> = ({ value, onChangeHandler, onKeyUpHandler }) => {
  return (
    <Input
      value={value}
      id="todo-input"
      className="todo-input"
      placeholder=""
      name={'todo-input'}
      label={''}
      title={'todo-input'}
      type={input.text}
      onChange={onChangeHandler}
      onKeyUp={onKeyUpHandler}
    >
      <span className="check__box" style={{ marginLeft: '0px', marginRight: '0px' }} />
    </Input>
  );
};

export default TodoInput;
