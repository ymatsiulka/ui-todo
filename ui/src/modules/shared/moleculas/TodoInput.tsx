import React from 'react';
import { Input } from 'modules/shared/atoms';
import { input } from 'appConstants';

interface TodoInputProps {
  value: string;
  onChangeHandler: React.ChangeEventHandler<HTMLInputElement>;
  onKeyUpHandler: React.KeyboardEventHandler<HTMLInputElement>;
}

const TodoInput: React.FC<TodoInputProps> = ({ value, onChangeHandler, onKeyUpHandler }) => {
  return (
    <Input
      value={value}
      id="todo-input"
      className="todo-input"
      placeholder=""
      name={'todo-input'}
      label={''}
      title={'todo-input'}
      type={input.text}
      onChange={onChangeHandler}
      onKeyUp={onKeyUpHandler}
    >
      <span className="check__box" style={{ marginLeft: '0px', marginRight: '0px' }} />
    </Input>
  );
};

export default TodoInput;
