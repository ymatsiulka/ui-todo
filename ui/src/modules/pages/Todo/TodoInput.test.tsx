import React from 'react';
import { render } from '@testing-library/react';
import TodoInput from './TodoInput';

jest.mock('modules/shared', () => {
  return {
    Input: jest.fn((props) => <div {...props} />),
  };
});

test('TodoInput should render properly', async () => {
  // Arrange
  // Act
  const { asFragment } = render(<TodoInput value="" onKeyUpHandler={jest.fn()} onChangeHandler={jest.fn()} />);

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
