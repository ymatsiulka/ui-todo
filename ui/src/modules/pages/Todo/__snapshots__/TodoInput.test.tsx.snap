// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly 1`] = `
<DocumentFragment>
  <div
    class="inputWrapper"
  >
    <span
      class="check__box"
      style="margin-left: 0px; margin-right: 0px;"
    />
    <input
      class="todo-input"
      id="todo-input"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`TodoInput should render properly 1`] = `
<DocumentFragment>
  <div
    class="inputWrapper"
  >
    <span
      class="check__box"
      style="margin-left: 0px; margin-right: 0px;"
    />
    <input
      class="todo-input"
      id="todo-input"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
